libcaf_net/caf/net/multiplexer.hpp

```diff
@@ -60,179 +31,81 @@ class CAF_NET_EXPORT multiplexer : public detail::atomic_ref_counted,
-
-  /// Returns the index of `mgr` in the pollset or `-1`.
-  ptrdiff_t index_of(const socket_manager_ptr& mgr) const noexcept;
-
-  /// Returns the index of `fd` in the pollset or `-1`.
-  ptrdiff_t index_of(socket fd) const noexcept;
+  virtual size_t num_socket_managers() const noexcept = 0;
 
   /// Returns the owning @ref middleman instance.
-  middleman& owner();
+  virtual middleman& owner() = 0;
 
   /// Returns the enclosing @ref actor_system.
-  actor_system& system();
-
-  // -- implementation of execution_context ------------------------------------
-
-  void ref_execution_context() const noexcept override;
-
-  void deref_execution_context() const noexcept override;
-
-  void schedule(action what) override;
-
-  void watch(disposable what) override;
+  virtual actor_system& system() = 0;
 
   // -- thread-safe signaling --------------------------------------------------
 
   /// Registers `mgr` for initialization in the multiplexer's thread.
   /// @thread-safe
-  void start(socket_manager_ptr mgr);
+  virtual void start(socket_manager_ptr mgr) = 0;
 
   /// Signals the multiplexer to initiate shutdown.
   /// @thread-safe
-  void shutdown();
+  virtual void shutdown() = 0;
 
   // -- callbacks for socket managers ------------------------------------------
 
   /// Registers `mgr` for read events.
-  void register_reading(socket_manager* mgr);
+  virtual void register_reading(socket_manager* mgr) = 0;
 
   /// Registers `mgr` for write events.
-  void register_writing(socket_manager* mgr);
+  virtual void register_writing(socket_manager* mgr) = 0;
 
   /// Deregisters `mgr` from read events.
-  void deregister_reading(socket_manager* mgr);
+  virtual void deregister_reading(socket_manager* mgr) = 0;
 
   /// Deregisters `mgr` from write events.
-  void deregister_writing(socket_manager* mgr);
+  virtual void deregister_writing(socket_manager* mgr) = 0;
 
   /// Deregisters @p mgr from read and write events.
-  void deregister(socket_manager* mgr);
+  virtual void deregister(socket_manager* mgr) = 0;
 
   /// Queries whether `mgr` is currently registered for reading.
-  bool is_reading(const socket_manager* mgr) const noexcept;
+  virtual bool is_reading(const socket_manager* mgr) const noexcept = 0;
 
   /// Queries whether `mgr` is currently registered for writing.
-  bool is_writing(const socket_manager* mgr) const noexcept;
+  virtual bool is_writing(const socket_manager* mgr) const noexcept = 0;
 
   // -- control flow -----------------------------------------------------------
 
   /// Polls I/O activity once and runs all socket event handlers that become
   /// ready as a result.
-  bool poll_once(bool blocking);
-
-  /// Runs `poll_once(false)` until it returns `true`.`
-  void poll();
+  virtual bool poll_once(bool blocking) = 0;
 
   /// Applies all pending updates.
-  void apply_updates();
+  virtual void apply_updates() = 0;
 
   /// Runs all pending actions.
-  void run_actions();
+  // virtual void run_actions() = 0;
```

Comment:
Please don't comment stuff out, just delete it. We have Git to bring it back if we ever need it.

Reply:
My mistake, I forgot to delete it.